src/Agent.Plugins/Artifact/FileShareProvider.cs

```diff
@@ -189,10 +193,48 @@ private async Task<FileShareDownloadResult> DownloadArtifactsAsync(ArtifactDownl
 
             sourcePath = sourcePath.TrimEnd(trimChars);
 
-            var artifactName =  new DirectoryInfo(destPath).Name;
+            var artifactName = new DirectoryInfo(sourcePath).Name;
 
-            IEnumerable<FileInfo> files =
-                new DirectoryInfo(sourcePath).EnumerateFiles("*", SearchOption.AllDirectories);
+            List<FileInfo> files =
+                new DirectoryInfo(sourcePath).EnumerateFiles("*", SearchOption.AllDirectories).ToList<FileInfo>();
+
+            // Getting list of file paths. It is useful to handle list of paths instead of files.
+            // Also it allows to use the same methods for FileContainerProvider and FileShareProvider.
+            List<string> paths = new List<string>();
+            foreach (FileInfo file in files)
+            {
+                string pathInArtifact = file.ToString().Remove(0, sourcePath.Length).TrimStart(trimChars);
```

Comment:
This is the same as the part of ApplyPatternsMapToFileShareItems method - could we move it to separate method?

Reply:
I think that I can move `file.ToString().Remove(0, sourcePath.Length).TrimStart(trimChars)` to separate method. But I'm not sure that foreach statement can be moved to separate method because foreach in DownloadFileShareArtifactAsync and foreach in ApplyPatternsMapToFileShareItems have some differences (foreach in ApplyPatternsMapToFileShareItems has additional commands).
@anatolybolshakov , is it ok to move 1 command `file.ToString().Remove(0, sourcePath.Length).TrimStart(trimChars)` to separate method?

Reply:
@denis-tikhomirov yes, since this is the same pattern - I think it make sense 👍

Reply:
Done. I added new method RemoveSourceDirFromPath to class ArtifactItemFilters